src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -2,46 +2,188 @@
+        if (index == size - 1) {
+            tail = node2;
+        }
+        size++;
     }
 
     @Override
-    public void addAll(List<T> list) {
+    public void addAll(List<V> list) {
+        for (int j = 0; j < list.size(); j++) {
+            add(list.get(j));
+        }
     }
 
     @Override
-    public T get(int index) {
-        return null;
+    public V get(int index) {
+        checkIndex(index);
+        if (index == size - 1) {
+            return tail.value;
+        }
+        Node next = head;
+        int counter = 0;
+        while (counter < index) {
+            next = next.next;
+            counter++;
+        }
+        return next.value;
     }
 
     @Override
-    public T set(T value, int index) {
-        return null;
+    public V set(V value, int index) {
+        checkIndex(index);
+        Node node = getByIndex(index);
+        V currentValue = node.value;
+        node.value = value;
+        return currentValue;
     }
 
     @Override
-    public T remove(int index) {
-        return null;
+    public V remove(int index) {
+        checkIndex(index);
+        V toReturn;
+        if (size == 1) {
+            size = 0;
+            return head.value;
+        }
+        if (index == 0) {
+            (head.next).prev = null;
+            toReturn = head.value;
+            head = head.next;
+            size--;
+            return toReturn;
+        }
+        if (index == size - 1) {
+            (tail.prev).next = null;
+            toReturn = tail.value;
+            tail = tail.prev;
+            size--;
+            return toReturn;
+        }
+        Node toRemove = getByIndex(index);
+        (toRemove.prev).next = toRemove.next;
+        (toRemove.next).prev = toRemove.prev;
+        size--;
+        return toRemove.value;
     }
 
     @Override
-    public boolean remove(T object) {
+    public boolean remove(V object) {
+        Node go = head;
+        int counter = 0;
+        while (!(go == null)) {
+            if (go.value == object || (go.value != null && go.value.equals(object))) {
+                V object2 = remove(counter);
+                return (object2 == object || (object2 != null && object2.equals(object)));
+            }
+            counter++;
+            go = go.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node getByIndex(int index) {
+        if (index == 0) {
+            return head;
+        }
+        if (index == size - 1) {
+            return tail;
+        }
+        if (size % index == 0 || index < size / 2) {
+            Node next = head;
+            for (int t = 1; t <= index; t++) {
+                next = next.next;
+                if (t == index) {
+                    return next;
+                }
+            }
```

Comment:
```suggestion
            Node currentNode = head;
            for (int t = 0; t < index; t++) {
                currentNode = currentNode.next;
            }
            return currentNode;
```

За аналогією спробуй змінити пошук з хвоста